src/data/http/http.ts

```diff
@@ -0,0 +1,42 @@
+import axios from 'axios';
+import { refreshToken as refreshTokenService } from '../auth/authServices';
+
+const axiosInstance = axios.create({
+  baseURL: 'http://173.212.214.70:3001/auth/',
+});
+
+axiosInstance.interceptors.request.use(
+  (config) => {
+    return {
+      ...config,
+      headers: {
+        ...config.headers,
+        Authorization: `Baerer ${localStorage.getItem('token')}`,
+      },
+    };
+  },
+  (error) => {
+    return Promise.reject(error);
+  },
+);
+
+axiosInstance.interceptors.response.use(
+  (config) => {
+    return config;
+  },
+  async (error) => {
+    if (!axios.isAxiosError(error)) {
+      return Promise.reject(error);
+    }
+    const refreshToken = localStorage.getItem('refreshToken');
+    if (!refreshToken || error.response?.status === 401) {
+      return Promise.reject(error);
+    }
+    const res = await refreshTokenService({ refreshToken });
+    localStorage.setItem('token', res.token);
+    localStorage.setItem('refreshToken', res.refreshToken);
+    return axiosInstance(error.config);
+  },
+);
+
+export default axiosInstance;
```

Comment:
I don't like the default export because in the files where you need to import this module you need to think how to name the module correctly. And when you export some entities not using default export you always known the name of module.
So I suggest to export axiosInstance not using default export